src/applications/mhv/medical-records/components/LabsAndTests/EkgDetails.jsx

```diff
@@ -9,7 +11,68 @@ const EkgDetails = props => {
 
   const formattedDate = formatDateLong(record?.date);
 
-  const download = () => {};
+  const generateEkgDetails = async () => {
+    const pdfData = {
+      headerLeft: 'Roberts, Jesse',
+      headerRight: 'Date of birth: January 1, 1970',
+      footerLeft: `Report generated by My HealtheVet and VA on ${moment().format(
+        'LL',
+      )}`,
+      footerRight: 'Page %PAGE_NUMBER% of %TOTAL_PAGES%',
+      headerBanner: [
+        {
+          weight: 'normal',
+          text:
+            "If you're ever in crisis and need to talk to someone right away, call the Veterans Crisis line at ",
+        },
+        {
+          weight: 'bold',
+          text: '988',
+        },
+        {
+          weight: 'normal',
+          text: '. Then select 1.',
+        },
+      ],
+      title: `electrocardiogram: ${record.name} on ${formattedDate}`,
+      subject: 'VA Medical Record',
+      preface:
+        'Your VA electrocardiogram list may not be complete. If you have any questions about your information, visit the FAQs or contact your VA Health care team.',
+      results: {
+        items: [
+          {
+            items: [
+              {
+                title: 'Date',
+                value: moment(record.date).format('MMMM Do YYYY') || ' ',
+                inline: true,
+              },
+              {
+                title: 'Location',
+                value: record.facility || ' ',
+                inline: true,
+              },
+              {
+                title: 'Provider',
+                value: record.orderedBy || ' ',
+                inline: true,
+              },
+            ],
+          },
+        ],
+      },
+    };
+
+    try {
+      await generatePdf('medicalRecords', 'electrocardiogram_report', pdfData);
+    } catch (error) {
+      // Error logging/presentation goes here...
```

Comment:
Let's make sure that any errors are getting reported to Sentry :-)

Reply:
addressed